code/edgeidentity/src/test/java/com/adobe/marketing/mobile/edge/identity/EventUtilsTests.java

```diff
@@ -0,0 +1,137 @@
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.edge.identity;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+
+import com.adobe.marketing.mobile.Event;
+import java.util.HashMap;
+import org.junit.Test;
+
+@SuppressWarnings("unchecked")
+public class EventUtilsTests {
+
+	// isAdIdEvent
+	@Test
+	public void test_isAdIdEvent_whenIsAdIdEvent_thenTrue() {
+		final Event event = createGenericIdentityEvent(
+			new HashMap<String, Object>() {
+				{
+					put(IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER, "adId");
+				}
+			}
+		);
+
+		assertTrue(EventUtils.isAdIdEvent(event));
+	}
+
+	@Test
+	public void test_isAdIdEvent_whenIsNotAdIdEvent_thenFalse() {
+		final Event event = createGenericIdentityEvent(
+			new HashMap<String, Object>() {
+				{
+					put("someKey", "someValue");
+				}
+			}
+		);
+
+		assertFalse(EventUtils.isAdIdEvent(event));
+	}
+
+	// getAdId
+	@Test
+	public void test_getAdId_whenIsNotAdIdEvent_thenEmpty() {
+		final Event event = createGenericIdentityEvent(
+			new HashMap<String, Object>() {
+				{
+					put("someKey", "someValue");
+				}
+			}
+		);
+
+		assertEquals("", EventUtils.getAdId(event));
+	}
+
+	@Test
+	public void test_getAdId_whenAllZeros_thenEmpty() {
+		final Event event = createGenericIdentityEvent(
+			new HashMap<String, Object>() {
+				{
+					put(
+						IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER,
+						IdentityConstants.Default.ZERO_ADVERTISING_ID
+					);
+				}
+			}
+		);
+
+		assertEquals("", EventUtils.getAdId(event));
+	}
+
+	@Test
+	public void test_getAdId_whenNull_thenEmpty() {
+		final Event event = createGenericIdentityEvent(
+			new HashMap<String, Object>() {
+				{
+					put(IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER, null);
+				}
+			}
+		);
+
+		assertEquals("", EventUtils.getAdId(event));
+	}
+
+	@Test
+	public void test_getAdId_whenEmpty_thenEmpty() {
+		final Event event = createGenericIdentityEvent(
+			new HashMap<String, Object>() {
+				{
+					put(IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER, "");
+				}
+			}
+		);
+
+		assertEquals("", EventUtils.getAdId(event));
+	}
+
+	@Test
+	public void test_getAdId_whenValid_thenValid() {
+		final Event event = createGenericIdentityEvent(
+			new HashMap<String, Object>() {
+				{
+					put(IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER, "adId");
+				}
+			}
+		);
+
+		assertEquals("adId", EventUtils.getAdId(event));
+	}
```

Comment:
let's add one for test_getAdId_whenUnexpectedType_thenEmpty

Reply:
Great point! Updated to implement this test case